test/unit/lib/plugins/aws/deploy/lib/createStack.test.js

```diff
@@ -88,17 +88,27 @@ describe('createStack', () => {
       });
     });
 
-    it('should use CloudFormation service role ARN if it is specified', () => {
-      awsDeploy.serverless.service.provider.cfnRole = 'arn:aws:iam::123456789012:role/myrole';
+    it('should use CloudFormation service role ARN if it is specified', async () => {
+      awsDeploy.serverless.service.provider.cfnRole = 'arn:aws:iam::123456789012:role/role-b';
 
       const createStackStub = sandbox.stub(awsDeploy.provider, 'request').resolves();
       sandbox.stub(awsDeploy, 'monitorStack').resolves();
 
-      return awsDeploy.create().then(() => {
-        expect(createStackStub.args[0][2].RoleARN).to.equal(
-          'arn:aws:iam::123456789012:role/myrole'
-        );
-      });
+      await awsDeploy.create();
+      expect(createStackStub.args[0][2].RoleARN).to.equal('arn:aws:iam::123456789012:role/role-b');
+    });
+
+    it('should use iam.deploymentRole service role if set', async () => {
```

Comment:
I know all new tests should be using `runServerless`, however with `deploy` cli arg it seems to run for unreasonable 5-10 seconds (as in deploy/index.test.js). Not sure I should doing it or keeping this simple and consistent with the rest?

Another con of using `runServerless` for deploy here is massive map of stubs similar to https://github.com/serverless/serverless/blob/master/test/unit/lib/plugins/aws/deploy/index.test.js#L225-L281

That said I'm still happy to do it if you reckon it's worthwhile and related to the current change.

Reply:
Let's not extend the old tests in any way like here. We should be using `runServerless` for new tests. When testing e.g. `deploy`, you can use `lastLifecycleHookName`: https://github.com/serverless/test/blob/master/docs/run-serverless.md#lastlifecyclehookname to avoid running hooks that are not essential for your test. I know some of these runs might be longer, but the structure of current tests here is really fragile and makes it harder to refactor functionality which is why we prefer to use more of an "integration-based" approach to testing

Reply:
Converted tests to use runServerless however not sure what hook I should be using to make them faster than 10s

Reply:
I remember in some situations those runs being slow due to custom resources being involved (not sure if here it's the case). Still then fix was to simply stub `spawn` via [`shouldStubSpawn`](https://github.com/serverless/test/blob/master/docs/run-serverless.md#shouldstubspawn) option

Reply:
Thanks @runk for going the `runServerless` route. It would also be great to move these `runServerless`-based tests to a separate `describe` as mentioned in: https://github.com/serverless/serverless/blob/master/test/README.md It allows to easier asses which tests are using `runServerless` and how many of them are still using the "old way". Let's name the `describe` block after the name of this test file.

Reply:
Moved tests into separate describe. Trick with `shouldStubSpawn: true` didn't help with long test time though..

Reply:
Yeah, in this case that might not actually do the trick. I believe we need to investigate these tests separately and evaluate what's causing them to run so long, as we already have a few similar ones with longer running times.

Reply:
Just fyi: I believe I've managed to find the cause for long-running tests and fix is incoming: https://github.com/serverless/test/pull/75